lib/index.js

```diff
@@ -974,3 +984,23 @@ exports.shallow = function (source) {
 
     return target;
 };
+
+
+exports.promiseWrap = function (bind, method, args, transform) {
+
+    return new Promise((resolve, reject) => {
+
+        const callback = function (error, result) {
+
+            if (Array.isArray(transform)) {
+                result = exports.transform(Array.from(arguments), transform);
+                error ? reject(result) : resolve(result);
+            }
+            else {
+                error ? reject(error) : resolve(result);
```

Comment:
Should probably be something like `error instanceof Error ? reject(error) : resolve(error || result)` if we want all the cases covered.